enhancements/ingress/mutable-publishing-scope.md

```diff
@@ -0,0 +1,362 @@
+the scope indicated in `status.endpointPublishingStrategy.loadBalancer.scope`.
+
+Crucially, by default, the operator *never* deletes the Service as long as the
+IngressController is not deleted; the operator requires that the user perform
+the disruptive action of deleting the Service in order to complete the
+operation.  The only exception is if the
+`ingress.operator.openshift.io/auto-delete-load-balancer` annotation is set on
+the IngressController, in which case the operator **does** delete the Service.
+
+### Risks and Mitigations
+
+It is important that the operator not perform a disruptive action (such as
+deleting a service load-balancer) for an active IngressController.  For this
+reason, the operator itself **never** deletes the Service for an
+IngressController that has not itself been marked for deletion.
+
+It is also important that the user understand that deleting a service
+load-balancer is disruptive (specifically that it may interrupt ingress traffic
+and may cause the load balancer's address to change).  For this reason, the
+`Progressing` status condition clearly indicates that deleting the Service is
+disruptive and provides an option for reverting back to the previous state
+without causing disruption.
+
+## Design Details
+
+### Test Plan
+
+The controller that manages the IngressController Deployment and related
+resources has unit test coverage; for this enhancement, the unit tests are
+expanded to cover the additional functionality.
+
+The operator has end-to-end tests; for this enhancement, the following test is
+added:
+
+1. Create an IngressController with the "LoadBalancerService" endpoint publishing strategy type and "External" scope.
+2. Verify that the operator creates a Service annotated for external scope.
+3. Set the IngressController's endpoint publishing strategy's scope to "Internal".
+4. If the operator is running on AWS and IBM Cloud, verify that the IngressController reports `Progressing=True`.
+5. If the operator is running on Azure or GCP, verify that the Service is annotated for internal scope.
+6. Set the IngressController's endpoint publishing strategy's scope to "External".
+7. Verify that the IngressController reports `Progressing=False`.
+8. Verify that the Service is annotated for external scope.
+
+### Graduation Criteria
+
+N/A.
+
+#### Dev Preview -> Tech Preview
+
+N/A.  This feature will go directly to GA.
+
+#### Tech Preview -> GA
+
+N/A.  This feature will go directly to GA.
+
+#### Removing a deprecated feature
+
+N/A.  We do not plan to deprecate this feature.
+
+### Upgrade / Downgrade Strategy
+
+If the user modifies `spec.endpointPublishingStrategy.loadBalancer.scope` on a
+version of OpenShift without this enhancement and upgrades to a version of
+OpenShift with this enhancement, the operator annotates the Service or sets
+`Progressing=True` on the IngressController as appropriate.  Thus the operator
+may effectuate a latent scope change, but it does not delete the Service.
```

Comment:
How/Why will it not delete the service in this case?

Reply:
The how and why are the same in the upgrade case as in the normal case:  If the ingresscontroller and the service specify different scopes and we don't know that the platform supports scope changes _in situ_, then the operator does not delete the service because doing so could disrupt traffic.  The behavior on upgrade or downgrade follows from and is consistent with the logic described in the proposal and implementation details.  Should I add something to that effect to the enhancement text?  I sought to adhere to the enhancement format without being too verbose or repetitive.